distributed/utils_test.py

```diff
@@ -1903,6 +1882,7 @@ def _reconfigure():
 
     with dask.config.set(
         {
+            "local_directory": tempfile.gettempdir(),
```

Comment:
diskutils is 
- creating the directory https://github.com/dask/distributed/blob/ff250f2b24025a49da294a6a8a140d07250c740b/distributed/diskutils.py#L44-L51
- creating a bunch of lock files, e.g. [global and purge lock](https://github.com/dask/distributed/blob/ff250f2b24025a49da294a6a8a140d07250c740b/distributed/diskutils.py#L125-L126) and a directory specific lock https://github.com/dask/distributed/blob/ff250f2b24025a49da294a6a8a140d07250c740b/distributed/diskutils.py#L62
- Most importantly, it is also cleaning up the workspace again https://github.com/dask/distributed/blob/ff250f2b24025a49da294a6a8a140d07250c740b/distributed/diskutils.py#L93-L109 which may conflict with the `TemporaryDirectory` context manager

In a follow up step, I also think we can think of disabling [purging](https://github.com/dask/distributed/blob/ff250f2b24025a49da294a6a8a140d07250c740b/distributed/diskutils.py#L242) on CI since we don't really care about his but this may significantly slow us down (e.g. introduce a config option)

Reply:
The worker is initializing WorksSpace and directory here https://github.com/dask/distributed/blob/ff250f2b24025a49da294a6a8a140d07250c740b/distributed/worker.py#L570-L585
where we read the `temporary-directory` variable in. the `os.getcwd()` there should probably be replaced with a `tempfile.gettempdir()` as well